lib/fluent/plugin/in_amqp.rb

```diff
@@ -2,6 +2,16 @@
 require 'fluent/plugin/input'
 require 'fluent/plugin/parser'
 require 'bunny'
+require 'zlib'
+require 'stringio'
+
+def gzip_inflate(string)
+  gz = Zlib::GzipReader.new(StringIO.new(string))    
```

Comment:
This looks pretty solid. My only suggestion would be to rescue any `Zlib::GzipFile::Error` exceptions raised by the zlib library unless they're being caught higher up.